File: 24-additive-prime/additive_prime.py
# Python program to calculate additive primes less than 500
# Written by M.V.Harish Kumar on 21-05-2024
def is_prime(n):
    if n == 2:
        return True

    for i in range(2,int(n**0.5)+1):
        if n % i == 0:
            return False

    return n > 1

print('Additive primes less than 500:')
n_primes = 0
for i in range(2, 500):
    sum_digits = sum([int(d) for d in str(i)])
    if is_prime(i) and is_prime(sum_digits):
        n_primes += 1
        print(i)
print(f"No. of additive primes less than 500: {n_primes}")

File: 24-additive-prime/additive_prime.go
// Golang program to calculate additive primes under 500
// Written by M.V.Harish Kumar on 21-05-2024
package main

import (
	"fmt"
	"math"
)

func isPrime(n int) bool {
	if n == 2 {
		return true
	}
	sqrt_num := int(math.Sqrt(float64(n)))
	for i := 2; i <= sqrt_num; i++ {
		if n%i == 0 {
			return false
		}
	}

	return n > 1
}

func digitSum(n int) int {
	sum := 0
	for n > 0 {
		sum += n % 10
		n /= 10
	}
	return sum
}

func main() {
	fmt.Println("Additive primes less than 500:")
	n_primes := 0
	for i := 2; i < 500; i++ {
		if isPrime(i) && isPrime(digitSum(i)) {
			fmt.Println(i)
			n_primes += 1
		}
	}
	fmt.Println("No. of additive primes less than 500:", n_primes)
}
